config/backend-config/backend-config.go

```diff
@@ -217,6 +217,13 @@ type CommonBackendConfig struct {
 	ctx              context.Context
 	cancel           context.CancelFunc
 	blockChan        chan struct{}
+	once             sync.Once
+}
+
+func (bc *CommonBackendConfig) init() {
+	bc.once.Do(func() {
+		bc.eb = pubsub.NewPublishSubscriber(context.TODO())
```

Comment:
Being this a dependency, shouldn't be passed/created when the backend config is created? This way it would be easier to mock and we will also avoid locking on a mutex each time we call `StartWithIDs` (see `Once.Do()`). Thoughts?

Reply:
This is a temporary solution, this needs to be refactored more. cc: @lvrach

Reply:
[Related ticket](https://www.notion.so/rudderstacks/Refactor-Common-Backend-Config-to-cater-Publish-Subscriber-in-a-better-way-and-make-it-more-compatib-0da64a6b37974b69ade1a8e583f01d4d)

Reply:
Agree with @fracasula, we can do it with the refactoring.